src/Cake.Core/Scripting/Processors/UriDirectiveProcessor.cs

```diff
@@ -60,14 +60,17 @@ public sealed override bool Process(IScriptAnalyzerContext context, string line,
         private Uri ParseUriFromTokens(string[] tokens)
         {
             Uri uri;
-            if (!Uri.TryCreate(tokens[1].UnQuote(), UriKind.Absolute, out uri))
+
+            var expandedTokens = tokens.Select(t => Environment.ExpandEnvironmentVariables(t));
```

Comment:
Perhaps we could add an `Expand` method to `ICakeEnvironment` instead of using System.Environment directly? That way it should be pretty straight forward mocking it during tests.

Reply:
There is the `GetEnvironmentVariables` function in the `ICakeEnvironment`. Wouldn't it be better to use that method and do the variable substitution manually?

Reply:
@Kemyke It would be OK to call `Environment.ExpandEnvironmentVariables(t)` from within the `CakeEnvironment` class. Putting it there would give us a way of mocking the behavior in tests.

Reply:
@patriksvensson  Yeah, i understand, i just look through the `ICakeEnvironment` interface and it looks a little bit redundant to me to have and `ExpandEnvironmentVariable` method and the `GetEnvironmentVariables` dictionary. You could do the mocking with that dictionary as well. 
It was just an idea to keep the `ICakeEnvironment` interface slim. I will do these changes and modify the PR.